scalyr_agent/__scalyr__.py

```diff
@@ -19,239 +19,168 @@
-        return parent_of_package_install
-    else:  # Installed using tarball or rpm/debian package
-        return os.path.dirname(parent_of_package_install)
 
+    # Those package types contain Scalyr Agent as frozen binary.
+    PACKAGE_INSTALL = "package"  # Indicates it was installed via a package manager such as RPM or Windows executable.
+    SOURCE_PACKAGE = "source_package"
+    TARBALL_INSTALL = "packageless"  # Indicates it was installed via a tarball.
 
-def __add_scalyr_package_to_path():
-    """Adds the path for the scalyr package and embedded third party packages to the PYTHONPATH.
+    # This type runs Scalyr Agent from the source code.
+    DEV_INSTALL = "dev"  # Indicates source code is running out of the original source tree, usually during dev testing.
 
-    If you add any new paths in this method, be sure to add them near the top of `setup.py` as well so as not
-    to break the Windows builds.
-    """
-    # prepend the third party directory first so it appears after the package root, third_party_pythonX
-    # and third_party_tls directories
-    sys.path.insert(0, os.path.join(get_package_root(), "third_party"))
 
-    if sys.version_info[0] == 2:
-        sys.path.insert(0, os.path.join(get_package_root(), "third_party_python2"))
+def __determine_platform():
+    system_name = platform.system().lower()
+    if system_name.startswith("win"):
+        return PlatformType.WINDOWS
+    elif system_name.startswith("linux"):
+        return PlatformType.LINUX
     else:
-        sys.path.insert(0, os.path.join(get_package_root(), "third_party_python3"))
+        return PlatformType.POSIX
 
-    # if we are not on windows, prepend the third party tls directory first so it appears after the package root
-    if not __is_frozen__ and (PY2_pre_279 or PY3_pre_32):
-        # Special case for backports module which is a multiple package module so we also need to
-        # add sub directory to pack when bundling a package and not installing it using setup.py
-        sys.path.insert(
-            0,
-            os.path.join(
-                get_package_root(), "third_party_tls/backports_ssl_match_hostname"
-            ),
-        )
 
-    sys.path.insert(0, os.path.dirname(get_package_root()))
+PLATFORM_TYPE = __determine_platform()
 
 
-def __determine_version():
-    """Returns the agent version number, read from the VERSION file."""
+def __determine_install_root_and_type() -> Tuple[str, InstallType]:
+    """
+    Determine the path for the install root and type of the installation.
+    """
 
-    file_names = ["VERSION"]
+    source_root = pl.Path(__file__).absolute().parent.parent
 
-    if __is_frozen__:
-        # also check for VERSION.txt file because there is a reserved filename - "VERSION" in Pyinstaller,
-        # and it expects that this file is a DLL.
-        file_names.append("VERSION.txt")
+    #
+    package_info_path = source_root / "scalyr_agent" / "package_info.json"
```

Comment:
It almost seems like we should also store build information in this file so we don't have two files (existing build_info + new package_info.json).